api/graphql/mutations/join_request.js

```diff
@@ -0,0 +1,32 @@
+export async function joinCommunity (communityId, userId) {
+  const user = await User.find(userId)
+  const community = await Community.find(communityId)
+
+  return GroupMembership.forPair(user, community, {includeInactive: true}).fetch()
+    .then(existingMembership => {
+      if (existingMembership) return existingMembership.get('active')
+        ? existingMembership
+        : existingMembership.save({active: true}, {patch: true}).then(membership => membership)
+      if (!!community) return user.joinCommunity(community).then(membership => membership)
+    })
+    .catch(error => ({error: error.message}))
+}
+
+export async function createJoinRequest (communityId, userId) {
+  return JoinRequest.create({
+    userId: userId,
+    communityId,
+  })
+  .then(request => ({ request }))
+}
+
+export async function acceptJoinRequest (joinRequestId, communityId, userId, moderatorId) {
```

Comment:
do we need to check somewhere that this is being called by the userId only?

Reply:
I added a couple checks for both user and community in the joinCommunity mutation, is that what you had in mind?